### ISSUE TITLE
`Any` not successfully typechecking for a specific type in a descriptor's `__set__` method

### ISSUE THREAD

**Issue by StockerMC** [microsoft/pyright#2213]
**Describe the bug**
When a descriptor's `__set__` method expects a specific type for the value being set other than `Any`, `Any` doesn't successfully typecheck for that value.

**Screenshots or Code**
```python
from typing import TypeVar, Any, overload, Type, Optional, Union

FV = TypeVar('FV', bound='flag_value')
F = TypeVar('F', bound='Flags')

class flag_value:
    def __init__(self, func):
        self.value: bool = bool(func(None))

    @overload
    def __get__(self: FV, instance: None, owner: Type[F]) -> FV:
        ...

    @overload
    def __get__(self, instance: F, owner: Type[F]) -> bool:
        ...

    def __get__(self: FV, instance: Optional[F], owner: Type[F]) -> Union[FV, bool]:
        if instance is None:
            return self

        return self.value

    def __set__(self, instance: 'Flags', value: int):
        self.value = bool(value)


class Flags:
    @flag_value
    def suppress(self):
        return 2

flags = Flags()
any_value: Any = 1
flags.suppress = any_value # this isn't fine

int_value: int = 1
flags.suppress = int_value # this is fine

union_value: Union[Any, int] = 1
flags.suppress = union_value # this is fine
```
```
error: Cannot assign member "suppress" for type "Flags"
```

**Expected behavior**
Setting the value of a descriptor when the value is `Any` when the descriptor expects a specific type should successfully typecheck

**VS Code extension or command-line**
I can reproduce this with the pyright vscode extension (v1.1.163), the pyright CLI tool (1.1.163) and the pylance vscode extension (v2021.8.2)

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.164, which I just published. It will also be in the next release of Pylance.
